### PR TITLE
DM-31234: Make script termination more robust, to avoid zombie processes.

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: python/lsst/ts/scriptqueue/queue_model.py [hunk 1]
```diff
@@ -625,7 +627,9 @@ def terminate_all(self):
         Returns
         -------
         info_list : `list` [`ScriptInfo`]
-            List of all scripts that were terminated.
+            List of all scripts that were terminated (except, possibly,
+            scripts that were terminated so quickly aftr starting that
```
Comment (wvreeven) [inline python/lsst/ts/scriptqueue/queue_model.py:631]: `aftr` should be `after`.

**Review by wvreeven — APPROVED** [review]
An elegant solution for a nasty issue. My only concern is the overloaded use of the verb `terminate`. Sentences like `terminate_all now terminates scripts that were terminated so soon after being started that the associated ScriptInfo never got the process info` makes me wonder why something, that already terminated, needs to be terminated. From the context it is clear, however, that zombie processes are meant.

**Comment by r-owen** [comment]
I agree that description was opaque. I have tried to rewrite it for clarity.

I also improved the termination process as follows, since your first review (thank you for agreeing to review it again). Now:
* ScriptInfo.terminate is now asynchronous and it now waits for the script process to start before terminating it. This should eliminate most zombie processes. However, they are still possible, if only because ScriptInfo.terminate has a time limit, to avoid hanging at shutdown.
* QueueModel.terminate_all is now asynchronous, since the synchronous version was not being used anywhere. it still kills zombie script processes, but now those should be rare (as just noted).
* QueueModel.wait_terminate_all is gone; use terminate_all instead.

**Review by wvreeven — APPROVED** [review]
Even more elegant than the previous implementation.

**Pull request merged**